webpack/assets/javascripts/react_app/common/MountingService.js

```diff
@@ -7,21 +7,49 @@ import componentRegistry from '../components/componentRegistry';
 
 export { default as registerReducer } from '../redux/reducers/registerReducer';
 
-const mountedNodes = [];
+/**
+ * mounted-nodes cache
+ */
+let mountedNodes = [];
 
 // In order to support turbolinks with react, need to
 // unmount all root components before turbolinks do the unload
 // TODO: remove it when migrating into (webpacker-react or react-rails)
-document.addEventListener('page:before-unload', () => {
-  let node = mountedNodes.shift();
+document.addEventListener('page:before-unload', () => unmountComponents());
+
+const unmountComponentsByNodes = (...nodes) => {
+  let node = nodes.shift();
 
   while (node) {
```

Comment:
any reason not to use `Array.prototype.forEach` here instead of manually removing items from the array one at a time?

Reply:
The origin of it was to be able to unmount the components while removing the nodes from the array (was a const before this pr).

Anyway, I will update it to a normal forEach since it makes more sense now.